reactive/kubernetes_control_plane.py

```diff
@@ -2761,12 +2761,22 @@ def get_kube_system_pods_not_running():
     FailedToGetPodStatus if unable to determine pod status. This can
     occur when the api server is not currently running. On success,
     returns a list of pods that are not currently running
-    or an empty list if all are running."""
+    or an empty list if all are running, ignoring pods whose names
+    start with those provided in the ignore-kube-system-pods config option."""
 
     result = get_pods("kube-system")
     if result is None:
         raise FailedToGetPodStatus
 
+    # Remove pods whose names start with ones provided in the ignore list
+    pod_names_space_separated = hookenv.config("ignore-kube-system-pods")
+    ignore_list = pod_names_space_separated.strip().split()
+    result["items"] = [
+        pod
+        for pod in result["items"]
+        if not any(pod["metadata"]["name"].startswith(name) for name in ignore_list)
```

Comment:
using `startswith` to cover pod deployments that have the pod name with random chars on the end